core/analysis.go

```diff
@@ -528,3 +531,45 @@ func AnalyzeRepositories(sess *Session) {
 	wg.Wait()
 
 }
+
+// WriteOutput will write a csv or json file to disk based on user specified options containing discovered findings
+func WriteOutput(sess *Session) {
+	if len(sess.Findings) > 0 {
+		if len(sess.OutputFile) != 0 {
+			f, err := os.Create(sess.OutputFile)
+			if err != nil {
+				sess.Out.Error("Failed writing to csv with error:\n%s\n", err)
+			}
+			defer f.Close()
+			if sess.CSV == true {
+				sess.Out.Info("Writing results in CSV format to: %s\n", sess.OutputFile)
+				w := csv.NewWriter(f)
+				defer w.Flush()
+
+				fields := sess.Findings[0].getFieldNames()
+				w.Write(fields)
+				for _, v := range sess.Findings {
+					_ = w.Write(v.getValues())
+				}
+				w.Flush()
+				f.Close()
+			} else if sess.JSON == true {
+				sess.Out.Info("Writing results in JSON format to: %s\n", sess.OutputFile)
+				b, err := json.MarshalIndent(sess.Findings, "", "    ")
+				if err != nil {
+					sess.Out.Error("Encountered error marshaling findings to json: %s\n",err)
+					return
+				}
+				c := string(b)
+				f.WriteString(c)
+				f.Close()
+			} else {
```

Comment:
Float this up earlier, for example when setting the prefix but the json/csv bools are false. I wouldn't error on it as it does not matter, but when debugging do what you are doing, just do it sooner. This will reduce the branching of the code, make the function smaller and tighter, and make tests much easier to write and maintain.

@funkwhatyouheard

Reply:
Not sure I'm following on this one, just need some clarification.